### ISSUE TITLE
Windows on ARM toolchain

### ISSUE THREAD

**Issue by ViralBShah** [JuliaPackaging/BinaryBuilder.jl#1341]
I see [Windows on ARM](https://github.blog/news-insights/product-news/arm64-on-github-actions-powering-faster-more-efficient-build-systems/) in Github Actions now.

Would it be possible to get a toolchain going to build all our binaries on Windows/ARM, and that can then be the stepping stone for Julia itself.

**Comment by ViralBShah** [comment]
@giordano I thought you had mentioned somewhere that there was a decision to be made in terms of what toolchain to use, but I can't remember where you said it. I just thought it would be worth noting that here.

I tried Windows on ARM in UTM (which is basically a nicely packaged qemu) on mac, and it runs quite well.

`juliaup` picks the windows x64 binaries  and I am able to run them in emulation and even do matmul - all out of the box.

**Comment by giordano** [comment]
> I think it isn't entirely clear what toolchain we should use for that platform (I think someone suggested clang with mingw, not sure gcc is supported at all), while it was much more obvious what to do for aarch64-freebsd

_Originally posted by @giordano in https://github.com/JuliaPackaging/Yggdrasil/issues/9405#issuecomment-2343854508_

**Comment by ViralBShah** [comment]
On the [MingW page](https://www.mingw-w64.org/downloads/) it recommends: https://github.com/mstorsjo/llvm-mingw/releases, which seems to use LLVM 20.

**Comment by giordano** [comment]
Yeah, that's what I mentioned in the message above, I think it was @mkitti who suggested it first.

**Comment by mkitti** [comment]
I believe my recommendation was based on the msys2 environments:
https://www.msys2.org/docs/environments/

![Image](https://github.com/user-attachments/assets/690f3699-bd1c-4af6-822c-d2a8b810327d)

As you can see the only supported toolchain for aarch64 is clang. I will also note that the only supported C library is ucrt rather than msvcrt.

I also think that we should consider how we might move the x86_64 builds to use ucrt rather than msvcrt, especially if we start building for ARM on Windows. I acknowledge this will likely be quite messy.

**Comment by ViralBShah** [comment]
You have an issue open for UCRT, which I am referring to here: https://github.com/JuliaLang/julia/issues/44006
